onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.h

```diff
@@ -54,8 +57,9 @@ class T5DecoderSubgraph : public Subgraph {
     return first_present_output_index_;
   }
 
-  int HasHiddenStates() const {
-    return has_hidden_state_;
+  bool UseSequenceAsInputIds() const {
+    // Custom model use sequence as input ids for decoder subgraph input.
+    return ParseEnvironmentVariableWithDefault<bool>(kUseSequenceAsInputIds, false);
```

Comment:
Environment variable is for testing only. Not recommend for production.

Shall we detect it from subgraph input_ids shape, like ['batch_size', 1] vs ['batch_size', 'sequence_length']?

Reply:
sounds good